demos-europe/demosplan-core
# fix (DPLAN-11393): Add report entry on change iteration of phase

**Ticket:** https://demoseurope.youtrack.cloud/issue/DPLAN-11393/Durchgangsnummer-und-Anderungen-werden-nicht-im-Protokoll-berucksichtigt

- detect change of iteration (as phase change)
- get iteration numbers and enhance created report message by them
- adjust message on reading the reports by adding the additional data
- do all of these for external- as well as for internal- phases
- improve readability by adjustments of variable naming

### How to review/test
Change the iteration of a phase of a procedure and check the report entries (protocoll) of the procedure.
